### ISSUE TITLE
questions about the pre-processing of categorical features

### ISSUE THREAD

**Issue by guolinke** [catboost/benchmarks#2]
From the pdf, I found: 
![image](https://user-images.githubusercontent.com/16040950/28332243-b0ffc414-6c26-11e7-955a-6f675709c74c.png)

I want to ask:
1. Is this processing used in all tools ? If yes, why CatBoost is still much better when using the same dataset without original categorical features ?  (I thought the advantage of CatBoost is the Categorical feature.)
2. the range of ```k``` for c_ij and d_ij ? 

Thanks !

**Comment by nikitxskv** [comment]
Hello!

1. This processing was made only for XGBoost, LightGBM and H2O. We used the original dataset for CatBoost.
2. `k` lies in the range` [1, l]`, where `l` is the size of the training dataset.

**Comment by guolinke** [comment]
@nikitxskv Thanks, it is really helpful.

BTW, the h2o.gbm also can handle the native categorical features well. Did you compare with it ?

**Comment by nikitxskv** [comment]
@guolinke Yes, we compared but almost all the experiments were not completed, they failed with different errors. The completed experiments (with "binary" catfeature type) had bad quality.

**Comment by guolinke** [comment]
@nikitxskv Thanks very much .
